pkg/common/cns-lib/volume/manager.go

```diff
@@ -248,6 +258,21 @@ func (m *defaultManager) CreateVolume(ctx context.Context, spec *cnstypes.CnsVol
 		log.Error(msg)
 		return nil, errors.New(msg)
 	}
+	var datastoreURL string
+	volumeCreateResult := interface{}(taskResult).(*cnstypes.CnsVolumeCreateResult)
+	if volumeCreateResult.PlacementResults != nil {
+		datastoreMoRef := volumeCreateResult.PlacementResults[0].Datastore
```

Comment:
Why are we picking the placementResult at index 0? CNS API does not define that the placement result at index 0 has the datastore set.

Reply:
Fixed.